src/dashboard/src/components/administration/views.py

```diff
@@ -184,6 +190,146 @@ def storage(request):
     system_directory_description = 'Available storage'
     return render(request, 'administration/locations.html', locals())
 
+def usage(request):
+    usage_dirs = _usage_dirs()
+
+    context = {'usage_dirs': usage_dirs}
+    return render(request, 'administration/usage.html', context)
+
+def _usage_dirs(calculate_usage=True):
+    # Put spaces before directories contained by the spaces
+    #
+    # Description is optional, but either a path or a location purpose (used to
+    # look up the path) should be specified
+    #
+    # If only certain sudirectories within a path should be deleted, set
+    # 'subdirectories' to a list of them
+    dir_defs = (
+        ('shared', {
+            'path': helpers.get_client_config_value('sharedDirectoryMounted')
+        }),
+        ('dips', {
+            'description': 'DIP uploads',
+            'path': os.path.join('watchedDirectories', 'uploadedDIPs'),
+            'contained_by': 'shared'
+        }),
+        ('rejected', {
+            'description': 'Rejected',
+            'path': 'rejected',
+            'contained_by': 'shared'
+        }),
+        ('failed', {
+            'description': 'Failed',
+            'path': 'failed',
+            'contained_by': 'shared'
+        }),
+        ('tmp', {
+            'description': 'Temporary file storage',
+            'path': 'tmp',
+            'contained_by': 'shared'
+        })
+    )
+
+    dirs = collections.OrderedDict(dir_defs)
+
+    # Resolve location paths and make relative paths absolute
+    for _, dir_spec in dirs.iteritems():
+        if 'contained_by' in dir_spec:
+            # If contained, make path absolute
+            space = dir_spec['contained_by']
+            absolute_path = os.path.join(dirs[space]['path'], dir_spec['path'])
+            dir_spec['path'] = absolute_path
+
+            if calculate_usage:
+                dir_spec['size'] = dirs[space]['size']
+                dir_spec['used'] = _usage_get_directory_used_bytes(dir_spec['path'])
+        elif calculate_usage:
+            # Get size/usage of space
+            space_path = dir_spec['path']
+            dir_spec['size'] = _usage_check_directory_volume_size(space_path)
+            dir_spec['used'] = _usage_get_directory_used_bytes(space_path)
+
+    return dirs
+
+def _usage_check_directory_volume_size(path):
+    # Get volume size (in 512 byte blocks)
+    try:
+        output = subprocess.check_output(["df", path])
+
+        # Second line returns disk usage-related values
+        usage_summary = output.split("\n")[1]
+
+        # Split value by whitespace and size (in blocks)
+        size = usage_summary.split()[1]
+
+        return int(size) * 512
+    except Exception, e:
```

Comment:
The modern syntax for providing a name for an exception is [`except Exception as e`](http://stackoverflow.com/questions/5119751/in-python-whats-the-difference-between-except-exception-as-e-and-except-exc).  Also, catching `Exception` is generally a [bad practice](http://ischenko.blogspot.ca/2005/01/exception-based-code-antipatterns.html), as it silences all errors, not just the ones you're expecting.

I was thinking something like
```python
except (ValueError, IndexError, subprocess.CalledProcessError):  # Known errors that could be raised
    logger.exception('Error checking directory size for %s', path)  # logger.exception already logs the exception raised
    return 0
```